app/controllers/documents_controller.rb

```diff
@@ -15,18 +15,27 @@ def show
 
   def update
     document = Document.find(params[:id])
-    document.update_attributes(document_update_params(document))
+    base_path = PathGeneratorService.new.path(document, params[:document][:title])
+    document.update(document_update_params(document).merge(base_path: base_path))
     DocumentPublishingService.new.publish_draft(document)
     redirect_to document, notice: "Preview creation successful"
   rescue GdsApi::HTTPErrorResponse, SocketError => e
     Rails.logger.error(e)
     redirect_to document, alert: "Error creating preview"
   end
 
+  def generate_path
+    document = Document.find(params[:id])
+    base_path = PathGeneratorService.new.path(document, params[:title])
+    render json: { base_path: base_path, available: true } if base_path.present?
```

Comment:
What happens if !base_path.present?

Reply:
Ah don't think that scenario is possible now, I'll remove the nil check.